docs/_docs/maintaining/releasing-off-stable-branches.md

```diff
@@ -0,0 +1,63 @@
+---
+
+Apart from having releases cut from the default `master` branch, Jekyll Core may occasionally cut releases containing security patches and
+critical bug-fixes for older versions under maintenance. Such releases are cut from specially named branches, following the pattern
+`[x].[y]-stable` where `[x]` denotes semver-major-version and `[y]`, the semver-minor-version. For example, the branch `3.9-stable` refers to
+commits released as part of `jekyll-3.9.x` series.
+
+Co-ordinating a release off a `*-stable` branch is complicated mainly because the default branch has to inevitably reflect the release as well.
+
+## Requirements
+
+- The maintainer has to have **write-access** to both the concerned `*-stable` and `master` branches.
+- The maintainer needs to complete the task using their **local CLI program** instead of dispatching via GitHub Web UI.
+- The maintainer is abreast with the workflow to [release off `master`]({{ 'docs/maintaining/releasing-a-new-version/' | relative_url }}). The
+  procedure documented in the following section is an abridged adaptation of the workflow for `master`.
+- A release post has been drafted and **is awaiting publish to `master`** via an approved pull request.
+- Stable internet connection.
+
+## Trigger release workflow
+
+1. Ensure that you've **checked out the concerned `*-stable` branch** and is up-to-date with its counterpart at `jekyll/jekyll` at GitHub.
+2. Bump the `VERSION` string in `lib/jekyll/version.rb`.
+3. Update the **History document** as documented [here]({{ 'docs/maintaining/releasing-a-new-version/#update-the-history-document' | relative_url }}).<br/>
+   (**IMPORTANT: Do not run `rake site:generate` on the stable branch though**).
+4. Copy the entire History section pertaining to current release and paste into a new tab / window of your text-editor. We will use this
+   temporary snippet at a future stage.
+5. Commit changes to the version-file and History document with commit message `Release :gem: v[CURRENT_VERSION]`.
+6. Push commit to upstream remote `jekyll/jekyll` at GitHub.
+
+## Publish release post
+
+1. Ensure the `Release Gem` workflow has completed successfully.
+2. Merge release-post pull request to `master`.
+
+## Update default branch to reflect release off the stable branch
+
+1. Locally, check out `master` and ensure it is up-to-date with its remote counterpart at `jekyll/jekyll` at GitHub.
+2. Update History document using the snippet in the temporary tab / window created earlier. The various sections in the History document are
+   primarily in reverse chronological order and secondarily scoped to the semver-major-version. For example, a release section for `v3.9.2`
+   will be listed above the section for `v3.9.1` but under release sections for v4.x.
+   The snippet stashed earlier has to be injected into the correct location manually.
+3. Optionally, update `VERSION` string in `lib/jekyll/version.rb`. (*If existing version is lesser than latest version*).
```

Comment:
When would this be the case? When we branch off for a major release and we have release candidates on master and a stable branch for the latest minor release from the previous major release? E.g. 4.0.0.rc1 is on master so we're releasing from 3.9-stable but we want the Jekyll website to reflect a stable (non-RC) version?

Reply:
> When would this be the case?

It happens when we're patching mainstream series. In fact, it happens right now, actually.

The current `master` contains commits deemed to bump the minor version. Understandably, we cut patches for the  **4.2.x** series from the `4.2-stable` branch (even though **4.2.x** is the current mainstream series).
However, once `4.2-stable` advances to `v4.2.2` or `v4.2.3`, `master` cannot remain at (older / lesser than) `v4.2.1` since jekyllrb.com is deployed off the `master`.
So, we have to update the `version.rb` on `master` as well in a consecutive step.

But, if we were cutting a release for an older series, say, **3.9.x** or even **4.1.x**, then `version.rb` on `master` doesn't reed to be updated.